src/pextlib1.0/tracelib.c

```diff
@@ -614,10 +616,48 @@ static void dep_check(int sock, char *path) {
         answer(sock, "#");
     }
 
+    if (-1 == fs_cs) {
+        /*
+         * We need to initialize fs_cs.
+         * It's easier to do so when we actually have an existing file.
+         */
+
+        char *lowercase_path = malloc(strlen(path) + 1);
+        strlcpy(lowercase_path, path, strlen(path) + 1);
+
+        for (char *tmp_ptr = lowercase_path; *tmp_ptr; ++tmp_ptr) {
+            *tmp_ptr = tolower(*tmp_ptr);
+        }
+
+        struct stat path_stat, lowercase_path_stat;
+        if (-1 == lstat(path, &path_stat)) {
+            ui_error(interp, "unable to get information on file %s", path);
+            answer(sock, "#");
+        }
+
+        if (-1 == lstat(path, &lowercase_path_stat)) {
+            fs_cs = 1;
+        }
+        else {
+            /*
+             * Both files exists, but we must make sure they
+             * are really the same file.
+             */
+            if (path_stat.st_ino == lowercase_path_stat.st_ino) {
+                /* Truly case-insensitive! */
+                fs_cs = 0;
+            }
+            else {
+                /* Just similar-named, still a case-sensitive FS. */
+                fs_cs = 1;
+            }
+        }
+    }
```

Comment:
I don't like the idea of manually fiddling with the case here. Couldn't we use `getattrlist(2)` to do this, maybe using this:

```c
#include <stdio.h>
#include <string.h>
#include <sys/attr.h>
#include <sys/mount.h>
#include <sys/param.h>
#include <unistd.h>

typedef struct volcaps {
	u_int32_t size;
	vol_capabilities_attr_t volcaps;
} volcaps_t;

int fs_case_sensitive(char *path) {
	struct statfs f;
	struct attrlist attrlist;
	volcaps_t volcaps;

	if (-1 == statfs(path, &f)) {
		/* failed to get mountpoint for a file; maybe the file doesn't exist */
		return 1;
	}

	memset(&attrlist, 0, sizeof(attrlist));
	attrlist.bitmapcount = ATTR_BIT_MAP_COUNT;
	attrlist.volattr = ATTR_VOL_CAPABILITIES;

	if (-1 == getattrlist(f.f_mntonname, &attrlist, &volcaps, sizeof(volcaps), 0)) {
		/* we don't know anything, assume case sensitive */
		return 1;
	}

	if ((attrlist.volattr & ATTR_VOL_CAPABILITIES) == 0) {
		/* failed to get mountpoint capabilities */
		return 1;
	}
	if ((volcaps.volcaps.valid[VOL_CAPABILITIES_FORMAT] & VOL_CAP_FMT_CASE_SENSITIVE) == 0) {
		/* capabilities bit for case-sensitivity not valid */
		return 1;
	}
	return (volcaps.volcaps.capabilities[VOL_CAPABILITIES_FORMAT] & VOL_CAP_FMT_CASE_SENSITIVE) > 0;
}

```
and then just use its return value?